contracts/Account.sol

```diff
@@ -398,17 +447,28 @@ contract Account is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed, I
         }
 
         // If there is no CELO to lock up and vote with, return.
-        if (unlockedCeloForGroup == 0) {
+        if (celoToVoteForGroup == 0) {
             return;
         }
 
+        uint256 accountLockedUnvotingGold = getLockedGold().getAccountNonvotingLockedGold(
+            address(this)
+        );
+
+        // There might be some locked unvoting (revoked) gold from previous transfers
+        uint256 toLock = accountLockedUnvotingGold >= celoToVoteForGroup
```

Comment:
Can a test case be added, for when `activateAndVotes()` is called after Alice (with a specific trategy) transfers stCelo to Bob (with default strategy)?

It seems like having a transfer between users with different strategies could cause `toLock > 0` when there is actually no unlocked balance, because of how we are accounting the `scheduledVotes[group].toVote` after a `scheduledtransfer()`

Reply:
I added e2e test "end-to-end-allowed-strategy.test" with different kind of transfers - can you please check if it does cover this scenario ?